pvlib/constants.py

```diff
@@ -0,0 +1,20 @@
+"""
+Useful constants for photovoltaic modeling.
+
+Several constants are derived from scipy.constants v1.1.0, which are derived
+from standardized CODATA. See https://physics.nist.gov/cuu/Constants/.
+
+Units appear in the variable suffix to eliminate ambiguity.
+"""
+
+# elementary charge, from scipy.constants.value('elementary charge')
+elementary_charge_C = 1.6021766208e-19
+
+# Boltzmann constant, from scipy.constants.value('Boltzmann constant')
+boltzmann_J_per_K = 1.38064852e-23
+
+# Define standard test condition (STC) temperature in degrees Celsius
+T_stc_degC = 25.
+
+# Define standard test condition (STC) temperature in Kelvin
+T_stc_K = 273.15 + T_stc_degC
```

Comment:
Recommend we use the common nomenclature here: q for elementary charge `q_C`, `k_J_per_K`, and the common variable for cell temperature `Tcell`: `Tcell_STC_degC`, `Tcell_STC_degK`.  I'm not a fan of the unit suffixes but I'm open to it, if the majority wants.

Reply:
I can see that the shorter names would be better in the formulas. While I prefer the units included, I won't get hung up on units in the name as long as the units are in a comment next to the constant's definition, and if we have a centralized definition of the constants, then there is probably less opportunity for confusion.

Reply:
Am not a fan of the units either.  Why not actually import the values from scipy rather than copying?  (Not that they're likely to change, but still...)

Reply:
I reverted the scipy implementation because the "base" installation of pvlib doesn't demand scipy and changing that is a separate issue out of scope.

Reply:
Let's open a new Issue to address having a common set of constants across all of pvlib.

Reply:
https://github.com/pvlib/pvlib-python/issues/483

I made these constants local to `sdm.py` for now. I didn't change any names, but they should be updated when the above issue is addressed.